src/editor/Core/dune

```diff
@@ -13,4 +13,4 @@
         yojson
         ppx_deriving_yojson.runtime
     )
-    (preprocess (pps ppx_deriving_yojson)))
+    (preprocess (pps ppx_deriving_yojson ppx_deriving.show)))
```

Comment:
@bryphe I initially had `ppx_deriving.runtime` in the `libraries` section but that made no difference.

The exact error is
```
erMap.cmo -c -impl src/editor/Core/BufferMap.re.pp.ml)
File "_none_", line 1:
Error: Unbound value Buffers.pp
error: command failed: 'refmterr' 'dune' 'build' '--root'
'.' '-j4' (exited with 1)
esy-build-package: exiting with errors above...

```

Reply:
Thanks for pushing up the repro! I tested it out and was able to get it working. You need the following:
- Add `ppx_deriving.runtime` as a lib above
- Add `[@deriving show]` on `Buffer.t`
- Add `[@deriving show]` on `BufferMetadata.t`
- Add `[@deriving show]` on `buftype`

The problem is that, even though `[@deriving show]` is on `BufferMap.t`, when it generates the code, it doesn't know how to `show` the custom types inside - so you have to annotate every type used. Hope that helps!